dist/tools/vera++/scripts/rules/T003.tcl

```diff
@@ -1,23 +1,32 @@
 #!/usr/bin/tclsh
 # Some keywords should be followed by a single space
+# MOD: This script has been modified to ignore C++ keywords in C files.
 
-set keywords {
+set c_keywords {
     case
-    class
-    delete
     enum
-    explicit
     extern
     goto
-    new
     struct
     union
+}
+
+set cpp_keywords {
+    class
+    delete
+    explicit
+    new
     using
 }
 
-proc isKeyword {s} {
-    global keywords
-    return [expr [lsearch $keywords $s] != -1]
+proc isCKeyword {s} {
+    global c_keywords
+    return [expr [lsearch $c_keywords $s] != -1]
+}
+
+proc isCPPKeyword {s} {
+    global cpp_keywords
+    return [expr [lsearch $cpp_keywords $s] != -1]
 }
```

Comment:
Shouldn't this also check for C keywords?

Reply:
I manually defined the CPP keywords above. The trick is in the _if_ above: if the keyword is a _pure C_ kw or the file is C++ and tje keywords is a C++ kw_, then the token is marked

Reply:
I understood that as much. But C keywords are also C++ keywords, so I would have expected something like

```suggestion
proc isCPPKeyword {s} {
    global cpp_keywords
    return [expr [lsearch $cpp_keywords $s] != -1] || [isCKeyword $s]
}
```

Reply:
see below